File: business/business-oj/src/main/java/com/glowxq/oj/topic/enums/TopicInfoStatus.java
package com.glowxq.oj.topic.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 作业/比赛过期策略枚举
 * 描述时间范围外的提交和访问规则
 *
 * @author glowxq
 * @version 1.0
 * @date 2025/4/11
 */
@Getter
@AllArgsConstructor
public enum TopicInfoStatus implements BaseType {

    NotStarted("NotStarted", "未开始"),

    InProgress("InProgress", "进行中"),

    Completed("Completed", "已完成"),
    ;

    private final String code;

    private final String name;

    public static TopicInfoStatus matchCode(String code) {
        for (TopicInfoStatus timeRangeType : TopicInfoStatus.values()) {
            if (timeRangeType.getCode().equals(code)) {
                return timeRangeType;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/topic/TopicInfoStatus.ts
import type { BaseEnum } from '@/enums/base';

// 主题状态枚举接口
interface TopicInfoStatusEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 主题状态枚举
 * 描述主题的当前状态
 */
export const TopicInfoStatus: { [key: string]: TopicInfoStatusEnum } = {
  /**
   * 未开始
   */
  NOT_STARTED: {
    code: 'NotStarted',
    name: '未开始',
    tagType: 'info'
  },

  /**
   * 进行中
   */
  IN_PROGRESS: {
    code: 'InProgress',
    name: '进行中',
    tagType: 'success'
  },

  /**
   * 已完成
   */
  COMPLETED: {
    code: 'Completed',
    name: '已完成',
    tagType: 'warning'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): TopicInfoStatusEnum | null {
  return Object.values(TopicInfoStatus).find(item => item.code === code) || null;
} 